server/src/main/java/org/elasticsearch/rest/RestResponse.java

```diff
@@ -112,14 +112,28 @@ public RestResponse(RestChannel channel, RestStatus status, Exception e) throws
         this.status = status;
         ToXContent.Params params = channel.request();
         if (e != null) {
-            Supplier<?> messageSupplier = () -> String.format(
-                Locale.ROOT,
-                "path: %s, params: %s, status: %d",
-                channel.request().rawPath(),
-                channel.request().params(),
-                status.getStatus()
-            );
-            if (status.getStatus() < 500) {
+            Supplier<?> messageSupplier;
+            boolean nodeOrShardUnavailableException = ExceptionsHelper.isNodeOrShardUnavailableTypeException(e);
+            if (nodeOrShardUnavailableException) {
+                messageSupplier = () -> String.format(
+                    Locale.ROOT,
+                    "path: %s, params: %s, status: %d; error: %s %s",
+                    channel.request().rawPath(),
+                    channel.request().params(),
+                    status.getStatus(),
+                    e.getClass(),
```

Comment:
Is this useful to put in the message since the exception is already passed to the logger and the stacktrace is logged?

Reply:
Removed.